index.js

```diff
@@ -32,9 +28,18 @@ app.get('/todos', (request, response) => {
   });
 });
 
+app.get('/todos/:id',(request, response)=> {
+  connection.query('SELECT * FROM todos WHERE todoId = ?', [request.params.id], (error, data) => {
+    if (data[0] === undefined) {
+      return response.status(404).json({status: 404, data: error});
```

Comment:
Hi guys, again, beautiful conversation! Really love it.

Please checkout this [Wikipedia page](https://en.wikipedia.org/wiki/HTTP_404) on http 404.

To give you my answer:

If you are looking for *a specific resource*, for example `/todos/5` and there is no todos with `todoId === 5` in the DB, then what you are looking for is really not found and the response should be 404. 

On the other hand, if you search for something and nothing is found, the response code can be `200, because no results is a valid response for search. Do you see the difference between:

`/todos/5` and `/todos?q.title=green`